### ISSUE TITLE
Submitting search form (mini) with enter key fires event handlers bound by jquery twice

### ISSUE THREAD

**Issue by koenner01** [magento/magento2#13793]
When submitting the search form in the header with the enter key on the keyboard, event handlers that were bound to the form submit (through jQuery) are fired twice.

### Preconditions
1. PHP 7.0
2. CE 2.2 (also tested 2.1.11)

### Steps to reproduce
1. Create a javascript which adds an event handler to the form submit
```
$('#search_mini_form').on('submit', function(e) {
    console.log('listening for form submit');
});
```
2. Submit the search form in the header with a value by pressing the enter button

### Expected result
1. Our created listener is only fired once

### Actual result
1. Listener is fired twice

In module-search/view/frontend/web/form-mini.js the _onKeyDown function is adding a trigger to the enter key
```
case $.ui.keyCode.ENTER:
    this.searchForm.trigger('submit');
    break;
```
This trigger is causing the submit event to be fired twice

**Comment by koenner01** [comment]
workaround can be to use the vanilla javascript addEventListener functionality

**Comment by magento-engcom-team** [comment]
@koenner01, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by amjadm61** [comment]
I'm working on it at #dmcdindia

**Comment by magento-engcom-team** [comment]
@amjadm61 thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by sidolov** [comment]
Hi @koenner01. Thank you for your report. 
The issue has been fixed in magento/magento2#15340 by @amjadm61 in 2.2-develop branch
Related commit(s):
 - [153faa6e823e0aa4baa5400503ed8bc993c0c9f8](https://github.com/magento/magento2/commit/153faa6e823e0aa4baa5400503ed8bc993c0c9f8)

The fix will be available with the upcoming 2.2.5 release.

**Comment by sidolov** [comment]
Hi @koenner01. Thank you for your report. 
The issue has been fixed in magento/magento2#15696 by @vgelani in 2.3-develop branch
Related commit(s):
 - [04583ac165cd775864e2ace38c5196c12ffc4ae8](https://github.com/magento/magento2/commit/04583ac165cd775864e2ace38c5196c12ffc4ae8)

The fix will be available with the upcoming 2.3.0 release.

**Comment by sidolov** [comment]
Hi @koenner01. Thank you for your report. 
The issue has been fixed in magento/magento2#16281 by @vgelani in 2.1-develop branch
Related commit(s):
 - [5064cb838fe3173299bd9e3086657f2e3c7e2179](https://github.com/magento/magento2/commit/5064cb838fe3173299bd9e3086657f2e3c7e2179)

The fix will be available with the upcoming 2.1.15 release.
